File: lab2_task1/main.cs
using System;

public class Task1 {
    public static string CheckBoardBounds(int n, int m, string steps) {
        for (int startX = 1; startX <= n; startX++) {
            for (int startY = 1; startY <= m; startY++) {
                int x = startX;
                int y = startY;
                bool isValid = true;

                foreach (char step in steps) {
                    switch (step) {
                        case 'L': x--; break;
                        case 'R': x++; break;
                        case 'D': y--; break;
                        case 'U': y++; break;
                        default: return "Ошибка в значении ходов";
                    }

                    if (x < 1 || x > n || y < 1 || y > m) {
                        isValid = false;
                        break;
                    }
                }

                if (isValid) {
                    return $"({startX},{startY})";
                }
            } 
        }
        return "No";
    }
    
    public static void Main() {
        Console.WriteLine("Введите длину доски: ");
        int n = Int32.Parse(Console.ReadLine());
        
        Console.WriteLine("Введите ширину доски: ");
        int m = Int32.Parse(Console.ReadLine());
        
        Console.WriteLine("Введите ходы для фигуры (LRDU): ");
        string steps = Console.ReadLine();
        
        Console.WriteLine(CheckBoardBounds(n, m, steps));
    }
}

File: lab2_task1/main.swift
func checkBoardBounds(n: Int, m: Int, steps: String) -> String {
    for startX in 1...n {
        for startY in 1...m {
            var x = startX, y = startY
            var isValid = true
            for step in steps {
                switch step {
                case "L": x -= 1
                case "R": x += 1
                case "D": y -= 1
                case "U": y += 1
                default: return "Ошибка в ходах"
                }
                if x < 1 || x > n || y < 1 || y > m {
                    isValid = false
                    break
                }
            }
            if isValid {
                return "(\(startX),\(startY))"
            }
        }
    }
    return "No"
}

let n = Int(readLine()!)!
let m = Int(readLine()!)!
let steps = readLine()!.trimmingCharacters(in: .whitespaces)
print(checkBoardBounds(n: n, m: m, steps: steps))